### PR TITLE
Add docker

### LINKED ISSUE #4: Add docker
## Detailed Description
Add docker service and that pulls all data and saves to a folder (which could be s3)

## Context
This will be used for nowcasting service - but other people could also use this

### PR DESCRIPTION
**Pull request by peterdudfield**
# Pull Request

## Description

- Add docker file
- add workflow for building and testing docker file
- add workflow for deploying docker file on release

Fixes #4 

## How Has This Been Tested?

build docker file locally and ran it, data was saved to [s3](https://s3.console.aws.amazon.com/s3/buckets/nowcasting-nwp?region=eu-west-1&prefix=data/&showversions=false)

- [ ] No
- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [ ] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,8 +46,23 @@ python3 metofficedatahub/app.py --save-dir="s3://bucket/folder"
 which will download all the files from Weather DataHub, join them together into a xarray dataset, and then save them.
 
 ## Docker
+The application can be run using docker
 
-TODO
+You can pull the production docker image from docker hub using
+```bash
+docker pull TODO
```
Comment (peterdudfield) [inline README.md:53]: to do once uploaded to dockerhub

**Review by jacobbieker — APPROVED** [review]
Looks good! I think there might be a docker workflow in .github that might be useful? I belive then it might cut down on duplication, but if not, this looks good!

File: .github/workflows/release-docker.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+name: Build and Push Docker Images to Docker Hub and GitHub
```
[the remark this answers is not in the document]
Comment (peterdudfield) [reply .github/workflows/release-docker.yml:1]: I couldnt see it - https://github.com/openclimatefix/.github/tree/main/.github/workflows, but maybe it is somewhere else. 

Perhaps we can move it across, when we make a next one. I wanted to make sure this one worked first

[the remark this answers is not in the document]
Comment (peterdudfield) [reply .github/workflows/release-docker.yml:1]: theres an issue for it already - https://github.com/openclimatefix/.github/issues/8

**Comment by peterdudfield** [comment]
@all-contributors please add @jacobbieker for review

**Pull request merged**